crates/sequencer/src/state.rs

```diff
@@ -22,11 +27,33 @@ use starknet_api::{
 /// See [rustc-hash](https://crates.io/crates/rustc-hash) for more information.
 #[derive(Default)]
 pub struct State {
-    classes: FxHashMap<ClassHash, ContractClass>,
-    compiled_class_hashes: FxHashMap<ClassHash, CompiledClassHash>,
-    contracts: FxHashMap<ContractAddress, ClassHash>,
-    storage: FxHashMap<ContractStorageKey, StarkFelt>,
-    nonces: FxHashMap<ContractAddress, Nonce>,
+    pub classes: FxHashMap<ClassHash, ContractClass>,
+    pub compiled_class_hashes: FxHashMap<ClassHash, CompiledClassHash>,
+    pub contracts: FxHashMap<ContractAddress, ClassHash>,
+    pub storage: FxHashMap<ContractStorageKey, StarkFelt>,
+    pub nonces: FxHashMap<ContractAddress, Nonce>,
+}
+
+impl State {
+    /// This will serialize the current state, and will save it to a path
+    pub fn dump_state_to_file(&self, path: &PathBuf) -> Result<(), SerializationError> {
+        let serializable_state: SerializableState = self.into();
+        let serialized_state = serializable_state.dump_state()?;
+
+        fs::write(path, serialized_state).map_err(|error| SerializationError::IoError {
+            reason: format!("failed to write dump state to the path {:?}", path),
+            context: error,
+        })?;
+
+        Ok(())
+    }
+
+    /// This will read a dump from a file and initialize the state from it
+    pub fn load_state_from_file(path: &PathBuf) -> Result<Self, SerializationError> {
+        let serilizable_state = SerializableState::load_state(path)?;
```

Comment:
```suggestion
        let serializable_state = SerializableState::load_state(path)?;
```

Reply:
resolved via [this](https://github.com/kkrt-labs/ef-tests/pull/159/commits/536b155d1182f5c3353afda2d253ecdf3e71582a) commit.

Reply:
resolved [here](https://github.com/kkrt-labs/ef-tests/pull/159/commits/bd6c71691c5b4e0f5e457503cb27640f93f078ae).